### PR TITLE
Add external package to work with unstructured objects

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:
This PR adds a new package called `external` with the purpose of easing the work with unstructured objects. Specifically the changes move functions from the Machine Controller to a package and it adds a new function called `Clone` that will be used by the Machine Set to clone external references. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Related to #1036 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/controller/external/util.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+// Clone uses the client and the reference to create a new cloned object.
+func Clone(c client.Client, ref *corev1.ObjectReference, namespace string) (*unstructured.Unstructured, error) {
+	from, err := Get(c, ref, namespace)
+	if err != nil {
+		return nil, err
+	}
+	to := from.DeepCopy()
+	to.SetResourceVersion("")
+	to.SetOwnerReferences(nil)
+	to.SetName("")
```
Comment (vincepri) [inline pkg/controller/external/util.go]: @detiber Can you quickly double check that these are the only things we should reset on an object before creating a new copy of it?

Comment (detiber) [reply pkg/controller/external/util.go]: Would we also potentially need to worry about finalizers here as well? @ncdc

Comment (ncdc) [reply pkg/controller/external/util.go]: I'm 50-50 on this. The end result is probably the same either way. Also, I don't see this being called yet?

Comment (vincepri) [reply pkg/controller/external/util.go]: Not yet, I followed the code from the pivot. I added a few more resets, just in case

**Comment by vincepri** [comment]
@detiber @ncdc ptal

**Comment by ncdc** [comment]
/lgtm
/hold
holding for jason's review - please clear when ready

**Pull request merged**